apis/python/operator/src/lib.rs

```diff
@@ -1,8 +1,104 @@
-use std::borrow::Cow;
+use std::{borrow::Cow, sync::Arc};
 
-use dora_node_api::{Metadata, MetadataParameters};
+use arrow::pyarrow::PyArrowConvert;
+use dora_node_api::{Data, Event, Metadata, MetadataParameters};
 use eyre::{Context, Result};
-use pyo3::{prelude::*, types::PyDict};
+use pyo3::{
+    exceptions::PyLookupError,
+    prelude::*,
+    types::{PyBytes, PyDict},
+};
+
+#[pyclass]
+pub struct PyEvent {
+    event: Event,
+    data: Option<Arc<Data>>,
+}
+
+#[pymethods]
+impl PyEvent {
+    pub fn __getitem__(&self, key: &str, py: Python<'_>) -> PyResult<PyObject> {
+        let value = match key {
+            "type" => Some(self.ty().to_object(py)),
+            "id" => self.id().map(|v| v.to_object(py)),
+            "data" => self.data(py),
+            "value" => self.value(py)?,
+            "metadata" => self.metadata(py),
+            "error" => self.error().map(|v| v.to_object(py)),
+            other => {
+                return Err(PyLookupError::new_err(format!(
+                    "event has no property `{other}`"
+                )))
+            }
+        };
+        value.ok_or_else(|| PyLookupError::new_err(format!("event has no property `{key}`")))
```

Comment:
I don't think that we should fail on None `value` and `data`. It means that the user can't send a `None` messages. Returning an optional value seems more appropriate.

Reply:
Makes sense! Done in 3eef207.

Reply:
Ah, I just saw that you edited your comment and proposed returning empty arrays as alternative. In my commit, I used Option to allow to differentiate between messages with an empty array and messages without data. Is this ok with you?

Reply:
I think that you wanted to fail them in case we were looking up data for event that does not include data.

I think that we should return:
- None or Fail if the event shall not have any data.
- empty list ( array or byte) if the sender sent an empty list as to keep the consistency for the user.

Reply:
Sounds reasonable. I pushed 8ec4523 to remove the mapping of empty arrays to `None` when sending outputs.